src/client.py

```diff
@@ -218,13 +220,48 @@ def _parse_items(self, resp_content):
         results = []
         xml = ET.ElementTree(ET.fromstring(resp_content))
         items = xml.getroot().findall("channel/item")
-        log.info(f"Found {len(items)} items from response")
+        log.info(f"Found {len(items)} items from response!")
+
         for item in items:
             result = self._parse_item(item)
             if result is not None:
                 results.append(result)
 
         return results
+    
+
+    # if we didn't get a magnet uri, attempt to resolve the magnet uri.
+    # todo for some reason Elementum cannot resolve the link that gets proxied through Jackett.
+    #  So we will resolve it manually for Elementum for now.
+    #  In actuality, this should be fixed within Elementum
+    def asinc_magnet_resolve(self, results, p_dialog):
+        size = len(results)
+        prog_from, prog_to = 20, 90
+        p_dialog.update(prog_from, message = translation(32751).format(size))
+
+        failed, count = 0, 0
+        with concurrent.futures.ThreadPoolExecutor(max_workers= os.cpu_count()*10) as executor:
+            future_to_magnet = {executor.submit(torrent.get_magnet, res["uri"]): res for res in results}
+            for future in concurrent.futures.as_completed(future_to_magnet):
+                count += 1
+                p_dialog.update(round(prog_from + (prog_to-prog_from)*(count/size)))
+                res = future_to_magnet[future]
+                try:
+                    magnet = future.result()
+                except Exception as exc:
+                    log.warning('%r generated an exception: %s' % ('',exc))
```

Comment:
Use log formatting
```suggestion
                    log.warning('%r generated an exception: %s', '', exc)
```

Also why are you passing an empty string for `%r`?

Reply:
done